Fix helm-test canary_test.go textparse API for Loki 3.6.x before v3.6.12.
Upstream adopted the new prometheus/common textparse API in v3.6.12;
earlier 3.6.x still need this patch to compile helm-test.
https://github.com/grafana/loki/tree/v3.6.11/production/helm/loki/src/helm-test
https://github.com/prometheus/common/blob/master/model/textparse/interface.go

--- a/production/helm/loki/src/helm-test/canary_test.go	2026-02-26 08:48:51
+++ b/production/helm/loki/src/helm-test/canary_test.go	2026-05-26 10:00:00
@@ -129,7 +129,7 @@ func testResultCanary(t *testing.T, ctx context.Contex
 	body, err := io.ReadAll(rsp.Body)
 	require.NoError(t, err, "Failed to read response body")
 
-	p, err := textparse.New(body, rsp.Header.Get("Content-Type"), true, false, nil)
+	p, err := textparse.New(body, rsp.Header.Get("Content-Type"), labels.NewSymbolTable(), textparse.ParserOptions{})
 	require.NoError(t, err, "Failed to create Prometheus parser")
 
 	for {
@@ -142,8 +142,8 @@ func testResultCanary(t *testing.T, ctx context.Contex
 			continue
 		}
 
-		l := labels.Labels{}
-		p.Metric(&l)
+		var l labels.Labels
+		p.Labels(&l)
 
 		// Currently we aren't validating any labels, just the metric name, however this could be extended to do so.
 		name := l.Get(model.MetricNameLabel)
